test/logger_test.rb

```diff
@@ -334,6 +341,25 @@
     end
   end
 
+  describe 'configuring error response' do
+    it 'blows up by default' do
+      assert_raises RuntimeError do
+        @logger.debug(message: true)
+      end
+    end
+
+    it 'silently swallows errors when configured to do so' do
+      apm = Minitest::Mock.new
+      apm.expect(:notify_error, nil, ["Logging schema validation error for {:message=>true}"])
+
+      @logger.configure_validation_error_response do |msg|
+        apm.notify_error("Logging schema validation error for #{msg}")
+      end
+
+      @logger.debug({ message: true })
```

Comment:
It's not clear here that we're asserting that the call to logger.debug with what appears to be a valid JSON message, but actually doesn't conform to the given JSON schema, **doesn't** raise an exception when we've specified a block for `configure_validation_error_response` (and thus the test passes).

Basically at first glance it appears like you've forgotten to put an assertion in your test. 
This isn't a technique I'm fond of because it runs counter to the 'assign, act, assert' pattern.
My suggestion below is the minimum that you could do to make it more obvious, without changing the structure of the test.

```suggestion
      nonconformant = { message: true }
      @logger.debug(nonconformant) # if configure_validation_error_response is not set this throws
```